src/lib/mongoBackend/MongoCommonUpdate.cpp

```diff
@@ -67,6 +67,14 @@ using namespace mongo;
 
 /* ****************************************************************************
 *
+* stringFilterP - 
+*/
+extern __thread StringFilter* stringFilterP;
```

Comment:
What about merging the branch that you have prepared for the next PR in the branch of the current PR? That should "fix" the thread var stuff an ease reviewing all in the same "context". Otherwise, we should take note of these inter-PR comments and verify in the new PR, i.e. we have to remember that some bits in the current PR need to be checked in the next one (complex for the reviewer).